src/client.js

```diff
@@ -136,14 +136,29 @@ export default class StitchClient {
    */
   authenticate(providerType, options = {}) {
     // reuse existing auth if present
-    return this.auth.getAccessToken().then(accessToken => {
+    return Promise.all(
+      [
+        this.auth.getAccessToken(),
+        this.auth.getLoggedInProviderType()
+      ]
+    ).then(([accessToken, loggedInProviderType]) => {
       if (accessToken) {
-        return this.auth.authedId();
+        // TODO: `anon` should be enumerated once providers are maintain
+        // an interface
+        if (providerType === 'anon' && loggedInProviderType === 'anon') {
+          return false; // is authenticated, skip log in
+        }
+
+        return this.logout(); // will not be authenticated, continue log in
```

Comment:
[minor] If logout fails, then we will never authenticate. From ticket:
`the SDK should attempt to log out (failure should be swallowed), and logged in with the new provider`. We should have a test case for this.

Reply:
I'll add a test– the code above should work to spec.